File: cpp/hot100/5.乘最多水的容器.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i=0,j=height.size()-1;
        int ans=0;
        while(i!=j){
            int temp=min(height[j],height[i])*(j-i);
            ans=max(temp,ans);
            if(height[j]>height[i]) i++;
            else j--;
        }
        return ans;
    }
};

File: python/hot100/5.乘最多水的容器.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l,r=0,len(height)-1
        ans=0
        while l!=r:
            temp=min(height[l],height[r])*(r-l)
            ans=max(ans,temp)
            if height[l]<height[r]:
                l+=1
            else:
                r-=1
        return ans
